synapse/tools/pullfile.py

```diff
@@ -0,0 +1,70 @@
+'''
+Fetches file details from a cortex and then pulls the associated file from the axon given
+'''
+
+import sys
+import argparse
+import binascii
+
+import synapse.exc as s_exc
+import synapse.glob as s_glob
+import synapse.common as s_common
+import synapse.telepath as s_telepath
+
+import synapse.lib.output as s_output
+
+
+def main(argv, outp=None):
+    pars = setup()
+    opts = pars.parse_args(argv)
+
+    if outp is None:
+        outp = s_output.OutPut()
+
+    if opts.output is None:
+        opts.output = '.'
+
+    s_common.gendir(opts.output)
+
+    axon = s_telepath.openurl(opts.axon)
```

Comment:
`with s_telepath.openurl(opts.axon) as axon:` would be slightly nicer and preclude having to fini at the end.